GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -1250,15 +1253,17 @@ void OnRevisionReadCompleted()
             {
                 if (!firstRevisionReceived && !FilterIsApplied(inclBranchFilter: true))
                 {
-                    semaphoreCurrentCommit.Wait();
-
-                    // This has to happen on the UI thread
+                    // No revisions at all received
                     this.InvokeAsync(
                             () =>
                             {
+                                if (!firstRevisionReceived)
+                                {
+                                    semaphoreUpdateGrid.Wait(cancellationToken);
+                                }
```

Comment:
Should we really block the UI thread?

Reply:
Is this really changed, RevisionReader runs on the default thread?
If so, SwitchToMainThreadAsync() (and default) is needed in several other situations.

...and most of BuildPathFilter() can be changed to run on the default thread too...

Reply:
I pushed b3db43141cc2d2142d76e1186c423b4fde389b03 to a private branch, not tested enough